contribution-guide.md

```diff
@@ -0,0 +1,143 @@
+- [Setting up the developer environment](setting-up-the-developer-environment)
+    - [Setting up IntelliJ IDEA](#setting-up-intelliJ-idea)
+    - [Setting up Eclipse](#setting-up-eclipse)
+- [Contributing to Ballerina grammar](#contributing-to-ballerina-grammar)
+- [Submitting your contribution](#submitting-your-contribution)
+- [Accepting the Contributor License Agreement](#accepting-the-contributor-license-agreement)
+- [Proposing changes to Ballerina](#proposing-changes-to-ballerina)
+
+## Getting engaged
+
+- The [Ballerina-Dev Google Group](#https://groups.google.com/forum/#!forum/ballerina-dev) is the main Ballerina project discussion forum. Also, you can use [StackOverflow](#https://stackoverflow.com/questions/tagged/ballerina) for support, [Slack](#https://app.slack.com/client/T47EAEKB5/DF818FX9Q) for real-time communications.
+
+- **IMPORTANT:** Send an email to security@ballerina.io to report sensitive security-related issues. For details, see the <a href="https://ballerina.io/security/">security policy</a>.
+
+- Submitting a bug is just as important as contributing to code. Click the **Issues** tab of the below GitHub repos, click **New Issue**, fill out all sections in the issue template, and submit the bug/issue report.
+
+    >**Tip:** If you are unsure whether you have found a bug, search existing issues in GitHub and raise it in the [Ballerina-Dev Google Group](#https://groups.google.com/forum/#!forum/ballerina-dev).
+   - compiler, runtime, standard library, or tooling: <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo.
+   - language specification: <a href="https://github.com/ballerina-platform/ballerina-spec/issues">ballerina-spec</a> repo.
+   - website: <a href="https://github.com/ballerina-platform/ballerina-dev-website/issues">ballerina-dev-website</a> repo. 
+
+## Getting started
+
+1. Download [Ballerina](#https://ballerina.io) and go through the [learning resources](#https://ballerina.io/learn).
+2. Read the [Ballerina Code of Conduct](#https://ballerina.io/code-of-conduct).
+3. Download the Java SE Development Kit (JDK) version 8 from [Oracle](#http://www.oracle.com/technetwork/java/javase/downloads/index.html) or [OpenJDK](#http://openjdk.java.net/install/index.html) and install it.
+>**Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed JDK.
+4. Download [Node.js](#https://nodejs.org/en/download/) (version 8.9.x or the latest LTS release) and intsall it.
+5. Download [npm](#https://www.npmjs.com/get-npm) (version 5.6.0 or later) and install it.
+6. If you are using IntelliJ IDEA (version 2016.3 or later), download the [Ballerina extension for IntelliJ IDEA](#https://plugins.jetbrains.com/plugin/9520-ballerina) and install it.
+7. Install and set up the ANTLR 4 plugin based on your IDE to check and validate a grammar rule you wrote.
+8. Start with GitHub issues that can be fixed easily:
```

Comment:
IMO, the structure and the flow of this document is not correct. There are many issues like 

- We have repetitive content here (Ex: Setting up the developer environment etc) 

- subsections are not properly added.  

- prerequisits are not clear (Ex: Download locations of JDK etc) 
- CLA Agreement is moved to bottom

etc

The Original document ( https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md) has a better flow than the proposed one. So please revisit and correct them

Reply:
1. Changed the download locations of JDK as follows:

Download the Java SE Development Kit (JDK) version 8 from one of the following locations:
    - [Oracle](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
    - [OpenJDK](http://openjdk.java.net/install/index.html) and install it.

2. Renamed the "Getting Started" topic as "Setting up the Prerequisites".

3. Moved the instructions on installing the IntelliJ plugin from the "Setting up the Prerequisites" section to "Setting up IntelliJ".

4. 3. Moved the instructions on installing the ANTLR plugin from the "Setting up the Prerequisites" section to "Setting up IntelliJ".

5. Renamed the "Getting Engaged" topic as "Getting Started".

6.Moved the "Start with GitHub issues that can be fixed easily:" point to the "Getting started" section".

When we consider the order of how the doc is organised, submitting the contribution via a PR comes last and that's where the CLA Agreement is presented to the user. That was why it was added last to demonstrate the exact order, which a user should follow when contributing.